superset-frontend/src/views/App.tsx

```diff
@@ -43,26 +49,39 @@ const bootstrap = JSON.parse(container?.getAttribute('data-bootstrap') ?? '{}');
 const user = { ...bootstrap.user };
 const menu = { ...bootstrap.common.menu_data };
 const common = { ...bootstrap.common };
+let lastLocation: string;
 initFeatureFlags(bootstrap.common.feature_flags);
 
-const RootContextProviders: React.FC = ({ children }) => (
-  <ThemeProvider theme={theme}>
-    <ReduxProvider store={store}>
-      <DndProvider backend={HTML5Backend}>
-        <FlashProvider messages={common.flash_messages}>
-          <DynamicPluginProvider>
-            <QueryParamProvider
-              ReactRouterRoute={Route}
-              stringifyOptions={{ encode: false }}
-            >
-              {children}
-            </QueryParamProvider>
-          </DynamicPluginProvider>
-        </FlashProvider>
-      </DndProvider>
-    </ReduxProvider>
-  </ThemeProvider>
-);
+const RootContextProviders: React.FC = ({ children }) => {
```

Comment:
Instead of putting this in `RootContextProviders`, it might be more appropriate to put the logic into `App` itself----the logger is not exactly a provider.

Reply:
This logic needs to stay with _global_ route change. Do you mean to move this logic to root dashboard / list component?

Reply:
I mean move it to the `App` component in L86.

Reply:
we chatted a little bit offline. Based on current way we use `<Router>` and `<RootContextProviders>`, keep this logic inside is probably the best solution.